netbench-orchestrator/src/orchestrator/cli.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::orchestrator::{OrchError, OrchResult};
+use serde::Deserialize;
+use std::{
+    fs::File,
+    path::{Path, PathBuf},
+};
+
+#[derive(Clone, Debug)]
+pub struct OrchestratorConfig {
+    // netbench
+    netbench_scenario_filename: String,
+    netbench_scenario_filepath: PathBuf,
+
+    // cdk
+    pub cdk_config: CdkConfig,
+}
+
+// Used for parsing the scenario file generated by the s2n-netbench project
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct CdkConfig {
+    #[serde(rename(deserialize = "NetbenchInfraPrimaryProd"))]
+    resources: CdkResources,
+}
+
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct CdkResources {
```

Comment:
Could we add a bit more detail to some of these? I'm mostly interested in information like
- what is in the public vs private bucket
- I'm guess the cloudfront distribution just fronts the public bucket?
- What is the subnet. Like the part of the VPC where the netbench trials run? Presumably that would mean that things only run in a single subet? 
- What is the key-value scheme? Like is the key the name of a host?

Reply:
Added docs and here is the relevant code that will use it in the upcoming PRs: [subnet](https://github.com/aws/s2n-netbench/blob/ak-fullOrchImport/netbench-orchestrator/src/ec2_utils/networking.rs#L177-L193)